tests/e2e/vmgroups_test.go

```diff
@@ -431,3 +432,141 @@ func (vg *VmGroupTest) TestVmGroupVolumeClone(c *C) {
 
 	misc.LogTestEnd(c.TestName())
 }
+
+// test to verify volume is removed after user created vmgroup is removed with "--remove-volume" option
+// 1. create a user created vmgroup "vmgroup_test1" and add VMs to this vmgroup
+// 2. create two volumes for this vmgroup, one is on default datastore, and the other is on non default_datastore
+// 3. verify mount/unmount for those two volumes
+// 4. after unmount, verify volume is detached for those two volumes
+// 5. remove VMs from this vmgroup, and remove the vmgroup with "--remove-volume" option
+// 6. verify that volumes do not exist after removing vmgroup
+func (vg *VmGroupTest) TestVmgroupRemoveWithRemoveVol(c *C) {
+	misc.LogTestStart(c.TestName())
+
+	// Create a volume in vgTestVMgroup1 on default_datastore
+	vg.createVolumes(c, vg.volName1)
+
+	// Add create privilege for the second datastore
+	out, err := adminutils.AddCreateAccessForVMgroup(vg.config.EsxHost, vgTestVMgroup1, vg.config.Datastores[1])
+	c.Assert(err, IsNil, Commentf(out))
+
+	// Create another volume on second datastore
+	// need to use full name for volume name
+	vg.createVolumes(c, vg.volName2+"@"+vg.config.Datastores[1])
+
+	// Run a container and then remove it using the first volume
+	out, err = dockercli.ExecContainer(vg.config.DockerHosts[0], vg.volName1, vg.testContainer)
+	c.Assert(err, IsNil, Commentf(out))
+
+	// Status should be detached after removing the container
+	status := verification.VerifyDetachedStatus(vg.volName1, vg.config.Datastores[0], vg.config.DockerHosts[0], vg.config.EsxHost)
+	c.Assert(status, Equals, true, Commentf("Volume %s is not detached", vg.volName1))
+
+	// Run a container and then remove it using the first volume
+	out, err = dockercli.ExecContainer(vg.config.DockerHosts[0], vg.volName2+"@"+vg.config.Datastores[1], vg.testContainer)
+	c.Assert(err, IsNil, Commentf(out))
+
+	// Status should be detached after removing the container
+	status = verification.VerifyDetachedStatus(vg.volName2, vg.config.Datastores[1],
+		vg.config.DockerHosts[0], vg.config.EsxHost)
+	c.Assert(status, Equals, true, Commentf("Volume %s is not detached", vg.volName2+"@"+vg.config.Datastores[1]))
+
+	// remove VM1 and VM2 from vgTestVMgroup1 and then remove the vmgroup
+	vmList := vg.config.DockerHostNames[0] + "," + vg.config.DockerHostNames[1]
+	out, err = adminutils.RemoveVMFromVMgroup(vg.config.EsxHost, vgTestVMgroup1, vmList)
```

Comment:
Suggestion, can instead pass the vg.config.DockerHostNames to RemoveVMFromVMgroup() and let it do the creation of the vmlist to pass to the admin cli command. Basically, the test functions are simple and shouldn't have to create special args for the util functions.